### PR TITLE
Remove all instances of deriving Typeable

### PR DESCRIPTION
**Pull request by exorcist365**
### Description

https://github.com/xmonad/xmonad-contrib/issues/548

### Checklist

  - [x] I've read [CONTRIBUTING.md](https://github.com/xmonad/xmonad/blob/master/CONTRIBUTING.md)

  - [x] I've confirmed these changes don't belong in xmonad-contrib instead

  - [x] I tested my changes with my setup ~~[xmonad-testing](https://github.com/xmonad/xmonad-testing)~~

  - [x] I updated the `CHANGES.md` file

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 ## unknown (unknown)
 
+  * Remove all instances of `deriving Typeable`
```
Comment (slotThe) [inline CHANGES.md]: It might be worth mentioning that this is because GHC derives this automatically now; we're not actually removing any functionality

Comment (liskin) [reply CHANGES.md]: Do we really need a changelog entry here?

Comment (liskin) [reply CHANGES.md]: My point being: users don't care that we removed some unnecessary statements from our code. It's just noise. The only possibly relevant user-visible change is that we're no longer compatible with GHC < 8.0, but we likely broke that compat earlier elsewhere.

Comment (slotThe) [reply CHANGES.md]: Mh that's a good point.  We should just remove this then

**Comment by exorcist365** [comment]
By README I mean CHANGES.md, my bad.

**Comment by liskin** [comment]
Thanks!

**Pull request merged**